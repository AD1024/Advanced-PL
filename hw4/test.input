assume n >= 0;

x := 0;
y := n;
while y > 0
  invariant x + y = n
  invariant y >= y
{
  x := x + 1;
  y := y - 1
};
assert x = n
