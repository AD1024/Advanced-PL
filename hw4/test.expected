[("n", (Syntax.VInt 10))]
[("y", (Syntax.VInt 0)); ("x", (Syntax.VInt 10)); ("n", (Syntax.VInt 10))]
